src/mono/wasm/debugger/tests/debugger-test/debugger-test.cs

```diff
@@ -653,3 +653,46 @@ internal static void ApplyUpdate (System.Reflection.Assembly assm, int version)
         }
     }
 
+
+public class Something
+{
+    public string Name { get; set; }
+    public Something() => Name = "Same of something";
+    public override string ToString() => Name;
+}
+
+public class Foo
+{
+    public string Bar => Stuffs.First(x => x.Name.StartsWith('S')).Name;
+    public System.Collections.Generic.List<Something> Stuffs { get; } = Enumerable.Range(0, 10).Select(x => new Something()).ToList();
+    public string Lorem { get; set; } = "Safe";
+    public string Ipsum { get; set; } = "Side";
+    public Something What { get; } = new Something();
+    public int Bart()
+    {
+        int ret;
+        if (Lorem.StartsWith('S'))
+            ret = 0;
+        else
+            ret = 1;
+        return ret;
+    }
+    public static void RunBart()
+    {
+        Foo foo = new Foo();
+        foo.Bart();
+        foo.OtherBar();
+        Console.WriteLine(foo);
+    }
+    public void OtherBar()
+    {
+        var a = 1;
+        var b = 2;
+        var x = "Stew";
+        var y = "00.123";
+        var c = a + b == 3 || b + a == 2;
+        var d = TimeSpan.TryParseExact(y, @"ss\.fff", null, out var ts) && x.Contains('S');
```

Comment:
Should we also test async cases for this? like given `Task<bool> FooAsync() { ..}`:

`x.Contains('S') && await FooAsync()`, `await BarAsync() && await FooAsync()`?

And would it work fine with a new line before/after `&&`?

Reply:
It works fine with a new line before/after `&&`.
I'll add a test with await.

Reply:
Added a test with async.